src/DBus.vala

```diff
@@ -110,6 +110,15 @@ public class Notifications.Server : Object {
         } else {
             var notification = new Notifications.Notification (app_name, app_icon, summary, body, actions, hints);
 
+            // Silence "Automatic suspend. Suspending soon because of inactivity." notifications
+            // These values and hints are taken from gnome-settings-daemon source code
+            if (hints.contains ("desktop-entry") && ((string) hints.get ("desktop-entry")) == "gnome-power-panel" &&
+                hints.contains ("x-gnome-privacy-scope") && ((string) hints.get ("x-gnome-privacy-scope")) == "system" &&
+                notification.priority == GLib.NotificationPriority.HIGH &&
+                expire_timeout == 0) {
+                return id;
+            }
```

Comment:
looking at GSD code, this will silence the low battery notifications too. not sure if it's also desired.

Reply:
I added a check for app_icon, seems like for "Automatic suspend" notifications it's always empty.